python/pyspark/errors/exceptions/base.py

```diff
@@ -89,13 +91,28 @@ def getSqlState(self) -> Optional[str]:
         See Also
         --------
         :meth:`PySparkException.getErrorClass`
+        :meth:`PySparkException.getMessage`
         :meth:`PySparkException.getMessageParameters`
         """
         return None
 
+    def getMessage(self) -> str:
+        """
+        Returns full error message.
+
+        .. versionadded:: 4.0.0
+
+        See Also
+        --------
+        :meth:`PySparkException.getErrorClass`
+        :meth:`PySparkException.getMessageParameters`
+        :meth:`PySparkException.getSqlState`
+        """
+        return f"[{self.getErrorClass()}] {self._message}"
+
     def __str__(self) -> str:
         if self.getErrorClass() is not None:
-            return f"[{self.getErrorClass()}] {self._message}"
```

Comment:
Is this the same as JVM side? `Exception.toString` We should probably keep `__str__` as is.

Reply:
Yeah, actually we don't touch the behavior of `__str__` but just arrange the code places here so I believe it would be fine.